src/refactorings/react/extract-use-callback/extract-use-callback.test.ts

```diff
@@ -0,0 +1,89 @@
+import { InMemoryEditor } from "../../../editor/adapters/in-memory-editor";
+import { Code, ErrorReason } from "../../../editor/editor";
+import { testEach } from "../../../tests-helpers";
+import { extractUseCallback } from "./extract-use-callback";
+
+describe("Extract useCallback", () => {
+  testEach<{ code: Code; expected: Code }>(
+    "should extract useCallback from an inline function",
+    [
+      {
+        description: "arrow expression",
+        code: `
+function Bar() {
+  return <Foo onFoo={[cursor](e) => console.log(e)} />;
+}`,
+        expected: `
+function Bar() {
+  const onFoo = useCallback(e => console.log(e), []);
+  return <Foo onFoo={onFoo} />;
+}`
+      },
+      {
+        description: "function expression",
+        code: `
+function Bar() {
+  return (
+    <Foo onFoo={[cursor]function handleFoo() {
+      console.log(e);
+    }} />
+  );
+}`,
+        expected: `
+function Bar() {
+  const onFoo = useCallback(function handleFoo() {
+    console.log(e);
+  }, []);
+
+  return <Foo onFoo={onFoo} />;
+}`
+      },
+      {
+        description: "uses a unique name",
+        code: `
+function Bar({onFoo}) {
+  return <Foo onFoo={[cursor](e) => {console.log(e); onFoo(e); }} />;
+}`,
+        expected: `
+function Bar({onFoo}) {
+  const onFoo2 = useCallback(e => {console.log(e); onFoo(e); }, []);
```

Comment:
👏 for handling conflicting names like that!

And +1 for not iterating the counter for more than 10. That will be very, very, _very_ unlikely.